### ISSUE TITLE
Drop PolarSSL support

### ISSUE THREAD

**Issue by fmorgner** [umurmur/umurmur#104]
From some discussions (e.g #82) it seems that PolarSSL is dead. Are there any platforms remaining that make use of Polar or can we drop support entirely?

**Comment by fmorgner** [comment]
I have done some preliminary research and it seems like Synology is the only platform still using mbedtls 1.x (a.k.a **old** mbed/Polar). It would be great if anyone could comment on that. Do they still use our PolarSSL backend?

**Comment by concatime** [comment]
I just want to know if you will change to openssl, or just evolve to mdebtls (old polartls).

**Comment by C4K3** [comment]
If (when) polarssl support is dropped, openssl, mbedtls and gnutls would continue to be supported.

**Comment by concatime** [comment]
Oh, I was unclear, sry. I mean, which one will be the default?

**Comment by fmorgner** [comment]
I would prefer to move on to mbed as a default, since it feels like the natural choiceto me given it is polar's successor. Would there be any reason to choose another implementation as the default?

**Comment by concatime** [comment]
No, I'm totally fine with this choice, but since you need to render a private and a certificate key with openssl command line, why not make openssl default? Also, one question. Does mumble/umurmur support X25519 (OpenSSL 1.1 does) ? I'm using rsa:4096 right now.

**Comment by omor1** [comment]
> since you need to render a private and a certificate key with openssl command line

This isn't true, though it is certainly common for creating a self-signed certificate. Both mbed TLS and GnuTLS have their own command-line tools for generating and signing certificates that can be used instead.

> Does mumble/umurmur support X25519?

I'm fairly certain that all crypto is delegated to the underlying library, which means that it will be fine as long as a recent OpenSSL/mbed TLS/GnuTLS is used.

**Comment by troxor** [comment]
It looks like this was done with 77c0d42704f05714dd19a0d52a020efc4ec246ea
